app/controllers/catalog_controller.rb

```diff
@@ -5,6 +5,30 @@ class CatalogController < ApplicationController
 
   include Blacklight::Catalog
 
+  # Destroy a layer if user is admin and layer is found
+  def destroy
+    if current_user and current_user.admin
+      solr = Blacklight.default_index.connection
+      response = solr.get 'get', :params => { :id => params[:id] }
+      if (response["doc"] == nil)
+        flash[:error] = 'Sorry, the layer you are trying to delete does not exist.'
+        redirect_to root_path
```

Comment:
is root_path just the homepage? could we redirect to the view of that layer instead?

Reply:
cannot redirect to the layer which does not exist

Reply:
whoops, yes, we should do this on the other one